File: src/Elsa.Dashboard/Models/StoreConfig.cs
using System.Text.Json.Serialization;

namespace Elsa.Dashboard.Models
{
  public class StoreConfig
  {
    [JsonPropertyName("serverUrl")]
    public string ServerUrl { get; set; } = null!;
    [JsonPropertyName("audience")]
    public string Audience { get; set; } = null!;
    [JsonPropertyName("clientId")]
    public string ClientId { get; set; } = null!;
    [JsonPropertyName("domain")]
    public string Domain { get; set; } = null!;
    [JsonPropertyName("useRefreshTokens")]
    public bool UseRefreshTokens { get; set; }
    [JsonPropertyName("useRefreshTokensFallback")]
    public bool UseRefreshTokensFallback { get; set; }
    [JsonPropertyName("monacoLibPath")]
    public string MonacoLibPath { get; set; } = null!;
  }
}

File: src/Elsa.Dashboard.Angular/src/models/storeConfig.ts
import { Auth0Client } from "@auth0/auth0-spa-js";

export class StoreConfig {
  serverUrl: string = "";
  audience: string = "";
  clientId: string = "";
  domain: string = "";
  useRefreshTokens: boolean = true;
  useRefreshTokensFallback: boolean = true;
  monacoLibPath: string = "";
  dataDictionaryIntellisense: string = "";
  auth0Client: Auth0Client = null;
  basePath: string = "";
}
